porcupine/plugins/python_tools.py

```diff
@@ -36,13 +36,23 @@ def run_tool(tool: str, code: str, path: Path | None) -> str:
             [str(python), "-m", tool, "-"],
             check=True,
             stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
             cwd=(Path.home() if path is None else path.parent),
             input=code.encode("utf-8"),
         )
         return result.stdout.decode("utf-8")
     except Exception:
         log.exception(f"running {tool} failed")
-        messagebox.showerror(f"Running {tool} failed", traceback.format_exc())
+        if tool == "black":
+            messagebox.showerror(
+                f"Running {tool} failed",
+                # can't black's use emojis in tkinter :(
```

Comment:
The error message is:

```
_tkinter.TclError: character U+10000 is above the range (U+0000-U+FFFF) allowed by Tcl
```

So you can simply remove the bad characters, like this for example:
```python
tkinter_safe_string = "".join(char for char in black_output if ord(char) <= 0xFFFF)
```

There is also #432 which is mostly about editing files containing emojies.